### ISSUE TITLE
Superflous parens get left behind by refactorer

### ISSUE THREAD

**Issue by TheDan64** [immunant/c2rust#133]
For example, if you refactor `(*foo).bar` to remove the deref, you get `(foo).bar`. A paren expr is not in use here, so there's no reason for it to persist. Also, we can't rely on cargo fix to make this fix because it does not produce a warning.

**Comment by TheDan64** [comment]
After rewriting newlib, saw this monstrosity:
```rust
- if !(*ptr)._result.is_null() {
+ if !((((&mut *ptr.unwrap()))))._result.is_null() {
```

**Comment by kkysen** [comment]
Closing as `c2rust refactor` in its current form is no longer being developed.
